src/fluree/db/flake.cljc

```diff
@@ -270,133 +304,149 @@
 
 (defn cmp-meta
   "Meta will always be a map or nil, but can be searched using an integer to
-  perform effective range scans if needed.
-  i.e. (Integer/MIN_VALUE) to (Integer/MAX_VALUE) will always include all meta values."
+  perform effective range scans if needed. i.e. (Integer/MIN_VALUE)
+  to (Integer/MAX_VALUE) will always include all meta values."
   [m1 m2]
   (let [m1h (if (int? m1) m1 (hash m1))
         m2h (if (int? m2) m2 (hash m2))]
     #?(:clj (Integer/compare m1h m2h) :cljs (- m1h m2h))))
 
 
-(defn cmp-pred [p1 p2]
-  (if (and p1 p2)
-    #?(:clj (Long/compare p1 p2) :cljs (- p1 p2))
-    0))
-
 (defn cmp-long [l1 l2]
   (if (and l1 l2)
     #?(:clj (Long/compare l1 l2) :cljs (- l1 l2))
     0))
 
+(defn cmp-subj
+  "Comparator for subject values. The supplied values are reversed before the
+  comparison to account for the decreasing sort order of subjects"
+  [s1 s2]
+  (cmp-long s2 s1))
 
-(defn cmp-flakes-spot [^Flake f1, ^Flake f2]
-  (combine-cmp
-    (cmp-long (.-s f2) (.-s f1))                            ;; reversed
-    (cmp-pred (.-p f1) (.-p f2))
-    (cmp-val (.-o f1) (.-o f2))
-    (cmp-meta (.-m f1) (.-m f2))))
+(defn cmp-pred [p1 p2]
```

Comment:
Adding these named comparators really adds to readability :+1:  Same with the s p o t wrappers.